api/v1alpha1/openapi.yaml

```diff
@@ -2502,6 +2532,231 @@ components:
+    DeviceApplicationSummaryType:
+      enum:
+      - 'Healthy'
+      - 'Degraded'
+      - 'Running'
+      - 'Error'
+      - 'Unknown'
+      x-enum-varnames:
+      - 'ApplicationsHealthy'
+      - 'ApplicationsDegraded'
+      - 'ApplicationsRunning'
+      - 'ApplicationsError'
+      - 'ApplicationsUnknown'
+    DeviceApplicationStatusType:
+      type: string
+      enum:
+      - 'Preparing'
+      - 'Starting'
+      - 'Running'
+      - 'Error'
+      - 'Unknown'
+      - 'Completed'
+      x-enum-varnames:
+      - 'ApplicationPreparing'
+      - 'ApplicationStarting'
+      - 'ApplicationRunning'
+      - 'ApplicationError'
+      - 'ApplicationUnknown'
+      - 'ApplicationCompleted'
+    DeviceRuntimeStatus:
+      type: object
+      properties:
+        status:
+          $ref: '#/components/schemas/DeviceRuntimeSummaryStatusType'
+          description: 'Status of the device runtime.'
+        info:
+          type: string
+          description: 'Human readable information indicating details about last transition.'
+    DeviceRuntimeSummaryStatusType:
+      type: string
+      enum:
+      - 'Online'
+      - 'Degraded'
+      - 'Error'
+      - 'Rebooting'
+      - 'PoweredOff'
+      - 'Unknown'
+      x-enum-varnames:
+      - 'DeviceRuntimeOnline'
+      - 'DeviceRuntimeDegraded'
+      - 'DeviceRuntimeError'
+      - 'DeviceRuntimeRebooting'
+      - 'DeviceRuntimePoweredOff'
+      - 'DeviceRuntimeUnknown'
+    DeviceUpdateStatus:
+      type: object
+      properties:
+        status:
+          $ref: '#/components/schemas/DeviceUpdateSummaryStatus'
+          description: 'Status of the device update.'
+        phase:
+          $ref: '#/components/schemas/DeviceUpdatePhaseStatus'
+          description: 'The passphrase for the private key.'
+    DeviceUpdateSummaryStatus:
+      type: object
+      properties:
+        status:
+          $ref: '#/components/schemas/DeviceUpdateStatusType'
+          description: 'Status summary type of the device update.'
+        info:
+          type: string
+          description: 'Human readable information about the last update transition.'
+    DeviceUpdateStatusType:
+      type: string
+      enum:
+      - 'UpToDate'
+      - 'OutOfDate'
+      - 'Updating'
+      - 'Unknown'
+      x-enum-varnames:
+      - 'DeviceUpdateUpToDate'
+      - 'DeviceUpdateOutOfDate'
+      - 'DeviceUpdateUpdating'
+      - 'DeviceUpdateUnknown'
+    DeviceUpdatePhaseStatus:
+      type: object
+      properties:
+        status:
+          $ref: '#/components/schemas/DeviceUpdatePhaseType'
+          description: 'Status of the device update phase.'
+        info:
+          type: string
+          description: 'Human readable information about the last update phase transition.'
+    DeviceUpdatePhaseType:
+      type: string
+      enum:
+      - 'Downloading'
+      - 'Verifying'
+      - 'Updating'
+      - 'Rebooting'
+      - 'RollingBack'
+      - 'Failed'
```

Comment:
- 'RolledBack' ? to indicate that the device has been automatically rolled back by the OS,.. and that may require action (you would see the device is booted to the previous image, and understand why that happens because of the 'RolledBack' status.

Reply:
Makes sense please update the DDD to reflect or provide edit access.

Reply:
What’s the diff between “RolledBack” and “Failed”?

BTW, I feel we need a condition with which the agent can aignal that some error occurred when trying to reconcile the spec (validation error, download error, rollback, …) and the agent will not try again until the spec has been updated.

Reply:
Failed could be: I could not finish download, there is a permanent problem with the image download.

Reply:
RollingBack is a very transient state "hopefully" and would also be overwritten with Rebooting. Failed is the canonical summary. Rolledback would be a reason.